src/Manager/Doctrine/ClientManager.php

```diff
@@ -7,45 +7,61 @@
 use Doctrine\ORM\EntityManagerInterface;
 use League\Bundle\OAuth2ServerBundle\Manager\ClientFilter;
 use League\Bundle\OAuth2ServerBundle\Manager\ClientManagerInterface;
-use League\Bundle\OAuth2ServerBundle\Model\Client;
+use League\Bundle\OAuth2ServerBundle\Model\ClientInterface;
 use League\Bundle\OAuth2ServerBundle\Model\Grant;
 use League\Bundle\OAuth2ServerBundle\Model\RedirectUri;
 use League\Bundle\OAuth2ServerBundle\Model\Scope;
 
 final class ClientManager implements ClientManagerInterface
 {
+    /**
+     * @var EntityManagerInterface
+     */
     private $entityManager;
 
-    public function __construct(EntityManagerInterface $entityManager)
+    /**
+     * @var class-string
+     */
+    private $clientFqcn;
+
+    /**
+     * @param class-string $clientFqcn
```

Comment:
```suggestion
     * @param class-string<ClientInterface> $clientFqcn
```

( meaning $clientFqnc needs to be a class-name of `ClientInterface` or a sub-type of that. )

after this, you should also be able to remove the `/** @var ClientInterface|null */` below since `EntityManager::getRepository` is generic ( `EntityManager::getRepository<T>(classname<T> $_): Repository<T>` ), and  so is `Repository::findOneBy` ( `Repository<T>::findOneBy(...): T|null` )

Reply:
Nice!